### PR TITLE
Fix failing signing functional tests

### PR DESCRIPTION
**Pull request by PatoBeltran**
Fix: https://github.com/NuGet/Home/issues/6302

Fix failing functional tests that was hanging because unexpected exception when shifting entries of a signed package archive.

### CODE REVIEW

**Review by emgarten — APPROVED** [review]

**Comment by rohit21agrawal** [comment]
Why was this timing out instead of just throwing ?

**Comment by PatoBeltran** [comment]
@rohit21agrawal I'm not sure why the call to `ReadAndWriteUntilPosition` was not throwing, but I could add a check to throw if `position > reader.BaseStream.Length`

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedArchiveTestUtility.cs [hunk 1]
```diff
@@ -211,7 +211,7 @@ private static Task ShiftSignatureMetadata(SigningSpecifications spec, BinaryRea
                 writer.Write(relativeOffsetOfLocalFileHeader);
 
                 // We already read and hash the whole header, skip only filenameLength + extraFieldLength + fileCommentLength (46 is the size of the header without those lengths)
-                ReadAndWriteUntilPosition(reader, writer, reader.BaseStream.Position + reader.BaseStream.Position + entry.HeaderSize - 46);
+                ReadAndWriteUntilPosition(reader, writer, reader.BaseStream.Position + entry.HeaderSize - 46);
```
Comment (rohit21agrawal) [inline test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignedArchiveTestUtility.cs:213]: could it the implementation of this function is running into an infinite while loop? if so, you should fix the loop.

**Comment by PatoBeltran** [comment]
@rohit21agrawal it seems like `reader.ReadBytes` does not throw when you try to read after the end of the stream, so I added a check to make sure the position is valid. The exception is not localized because it is a test.

**Comment by rohit21agrawal** [comment]
:shipit: once CI is green, and just make sure you don’t have something similar in product code

**Pull request merged**